cirq/testing/order_tester_test.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fractions
+import pytest
+
+from cirq.testing.order_tester import OrderTester
+
+def test_add_ordering_group_correct():
+    ot = OrderTester()
+    ot.add_ascending(-4, 0)
+    ot.add_ascending(1, 2)
+    ot.add_ascending_equivalence_group(fractions.Fraction(6, 2),
+                                       fractions.Fraction(12, 4), 3, 3.0)
+
+def test_add_ordering_group_incorrect():
+    ot = OrderTester()
+    ot.add_ascending(1, 2)
+    with pytest.raises(AssertionError):
+        ot.add_ascending(1, 3)  # not ascending w.r.t. previous call
+    with pytest.raises(AssertionError):
+        ot.add_ascending(6, 6)  # not ascending within call
+    with pytest.raises(AssertionError):
+        ot.add_ascending(99, 10)  # not ascending within call
+
+def test_add_ordering_equivalence_group_incorrect():
+    ot = OrderTester()
+    with pytest.raises(AssertionError):
+        ot.add_ascending_equivalence_group(1, 3)  # not an equivalence group
+    ot.add_ascending(1)
+    with pytest.raises(AssertionError):
+        ot.add_ascending_equivalence_group(0, 0.)  # not ascending w.r.t
+                                                   # previous items
+
```

Comment:
Test cases to include:

InconsistentWithOthers/AlwaysTrue/AlwaysFalse `__le__`/`__ge__`/`__lt__`/`__gt__`.

`__le__` or `__ge__` returns false when `__eq__` returns true.

`__lt__` and `__gt__` both return false when `__eq__` returns false.

Reply:
Can this be done in a follow-up commit?

Reply:
No, this is crucial functionality that must be work from the start.